custom/ewsghana/api.py

```diff
@@ -404,6 +404,13 @@ def sms_user_sync(self, ews_smsuser, **kwargs):
             return None
         sms_user.user_data['to'] = ews_smsuser.to
 
+        if ews_smsuser.role == 'facility_manager':
+            role = UserRole.by_domain_and_name(self.domain, 'Facility manager')
+            if role:
+                dm = sms_user.get_domain_membership(self.domain)
+                dm.role_id = role[0].get_id
+
+        sms_user.save()
```

Comment:
User need to be save before setting location because set_location method use _id to create mapping.